src/test/kotlin/uk/gov/justice/digital/hmpps/supportadditionalneedsapi/integration/resource/GetEhcpStatusTest.kt

```diff
@@ -0,0 +1,95 @@
+package uk.gov.justice.digital.hmpps.supportadditionalneedsapi.integration.resource
+
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import uk.gov.justice.digital.hmpps.supportadditionalneedsapi.integration.IntegrationTestBase
+import uk.gov.justice.digital.hmpps.supportadditionalneedsapi.randomValidPrisonNumber
+import uk.gov.justice.digital.hmpps.supportadditionalneedsapi.resource.model.EhcpStatusResponse
+import uk.gov.justice.digital.hmpps.supportadditionalneedsapi.resource.model.ErrorResponse
+import uk.gov.justice.digital.hmpps.supportadditionalneedsapi.resource.model.UpdateEhcpRequest
+
+class GetEhcpStatusTest : IntegrationTestBase() {
+  companion object {
+    private const val URI_TEMPLATE = "/profile/{prisonNumber}/ehcp-status"
+  }
+
+  @Test
+  fun `Get EHCP status for a given prisoner`() {
+    // Given
+    stubGetTokenFromHmppsAuth()
+    stubGetDisplayName("testuser")
+    val prisonNumber = randomValidPrisonNumber()
+    anElSPExists(prisonNumber)
+
+    // When
+    val response = webTestClient.get()
+      .uri(URI_TEMPLATE, prisonNumber)
+      .headers(setAuthorisation(roles = listOf("ROLE_SUPPORT_ADDITIONAL_NEEDS__ELSP__RO"), username = "testuser"))
+      .exchange()
+      .expectStatus()
+      .isOk
+      .returnResult(EhcpStatusResponse::class.java)
+
+    // Then
+    val actual = response.responseBody.blockFirst()
+    assertThat(actual).isNotNull()
+    assertThat(actual!!.hasCurrentEhcp).isTrue()
+  }
+
+  @Test
+  fun `Return error when EHCP status does not exist for prisoner`() {
+    // Given
+    stubGetTokenFromHmppsAuth()
+    stubGetDisplayName("testuser")
+    val prisonNumber = randomValidPrisonNumber()
+
+    // No EHCP status is created or saved for this prison number
+
+    // When
+    val response = webTestClient.get()
+      .uri(URI_TEMPLATE, prisonNumber)
+      .headers(setAuthorisation(roles = listOf("ROLE_SUPPORT_ADDITIONAL_NEEDS__ELSP__RO"), username = "testuser"))
+      .exchange()
+      .expectStatus()
+      .isNotFound
+      .returnResult(ErrorResponse::class.java)
+
+    // Then
+    val actual = response.responseBody.blockFirst()
+    assertThat(actual).isNotNull()
+    assertThat(actual!!.userMessage).isEqualTo("ELSP plan not found for prisoner [$prisonNumber]")
+  }
+
+  @Test
+  fun `Update EHCP status for a given prisoner`() {
```

Comment:
Minor; should this test be in a test class related to updates/edits rather then get ?
(I cant remember what we've done for the other int.tests in this repo? but I notice the class name is GetEhcpStatusTest 🤔 )

Reply:
yeah just being lazy... two tics..